File: 1011.cpp
#include <stdio.h>
#include <math.h>
int main() {
    int t, x, y, dis, times, max;
	scanf("%d", &t);
	for (; t > 0; t--) {
		scanf("%d %d", &x, &y);
		max = times = (int)sqrt(y - x);
		dis = (y - x) - (((1 + max) * max / 2) * 2 - max);
		times += (max - 1) + dis / max + (dis % max ? 1 : 0);
		printf("%d\n", times);
	}
    return 0;
}

File: 1011.py
import math
t = int(input())
for i in range(t):
    x, y = map(int, input().split())
    mx = ti = int(math.sqrt(y - x))
    dis = (y - x) - ((1 + mx) * mx - mx)
    ti += (mx - 1) + int(dis / mx) + (1 if dis % mx else 0) 
    print(ti)
